optuna/visualization/matplotlib/_parallel_coordinate.py

```diff
@@ -122,7 +122,7 @@ def _get_parallel_coordinate_plot(info: _ParallelCoordinateInfo) -> "Axes":
     segments = [np.column_stack([x, y]) for x, y in zip(xs, dims_obj_base)]
     lc = LineCollection(segments, cmap=cmap)
     lc.set_array(np.asarray(info.dim_objective.values))
-    axcb = fig.colorbar(lc, pad=0.1)
+    axcb = fig.colorbar(lc, pad=0.1, ax=ax)
```

Comment:
Apply the change suggested by `To silence this warning, explicitly pass the 'ax' argument to colorbar().`

```
$ pytest tests/visualization_tests/test_parallel_coordinate.py::test_plot_parallel_coordinate_customized_target_name
============================================================= test session starts =============================================================
platform darwin -- Python 3.9.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/nzw/Documents/optuna, configfile: pyproject.toml
plugins: anyio-3.5.0
collected 1 item

tests/visualization_tests/test_parallel_coordinate.py .                                                                                 [100%]

============================================================== warnings summary ===============================================================
tests/visualization_tests/test_parallel_coordinate.py::test_plot_parallel_coordinate_customized_target_name
  /opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/optuna-3.1.0.dev0-py3.9.egg/optuna/visualization/matplotlib/_parallel_coordinate.py:125: MatplotlibDeprecationWarning:

  Starting from Matplotlib 3.6, colorbar() will steal space from the mappable's axes, rather than from the current axes, to place the colorbar.  To silence this warning, explicitly pass the 'ax' argument to colorbar().

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```